### PR TITLE
tiltfile: add encode_yaml and encode_json

### PR DESCRIPTION
**Pull request by landism**
the complement of `decode_yaml` and `decode_json`, i.e., from a starlark object to a yaml/json-encoded string (or, really, a `Blob`, to make it easier to pass to `k8s_yaml`)

also: changed `decode_yaml` and `decode_json` to take a `Union[str|Blob]` instead of a `str`, since it's likely most of the inputs to those functions will be `Blob`s already (e.g., from the output of `read_file` or `local`)

I really wanted to name all these `yaml.load` / `yaml.loads` / `yaml.dumps`, but `load` is a keyword in starlark, making that impossible :(

### CODE REVIEW

File: internal/tiltfile/encoding/json_test.go [hunk 1]
```diff
@@ -118,3 +168,27 @@ test()
 	}
 	require.NoError(t, err)
 }
+
+func TestEncodeJSONNonStringMapKey(t *testing.T) {
```
Comment (jazzdan) [inline internal/tiltfile/encoding/json_test.go]: Is there a typo here and in the test below?

```suggestion
func TestEncodeJSONOnStringMapKey(t *testing.T) {
```

File: internal/tiltfile/encoding/json.go [hunk 2]
```diff
@@ -40,12 +41,22 @@ func readJSON(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple
...
 // reads json from a string
 func decodeJSON(thread *starlark.Thread, fn *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
-	var contents starlark.String
+	var contents starlark.Value
 	if err := starkit.UnpackArgs(thread, fn.Name(), args, kwargs, "json", &contents); err != nil {
 		return nil, err
 	}
 
-	return jsonStringToStarlark(contents.GoString(), "")
+	var s string
```
Comment (nicks) [inline internal/tiltfile/encoding/json.go]: this should be equivalent to value.AsString(s)?

File: internal/tiltfile/encoding/json_test.go (hunk 1 above)
Comment (landism) [reply internal/tiltfile/encoding/json_test.go]: nope! JSON only supports strings as map keys, and this test is for serializing a map that has a map key that's not a string and making sure returns an error that map keys must be strings.

File: internal/tiltfile/encoding/json.go (hunk 2 above)
Comment (landism) [reply internal/tiltfile/encoding/json.go]: oh cool, I looked for something to handle the blob/string union and was surprised I couldn't find it

**Pull request merged**